classes/mock/generator.php

```diff
@@ -490,11 +490,39 @@ protected function generateMethodSignature(\reflectionMethod $method)
 
     protected function generateClassCode(\reflectionClass $class, $mockNamespace, $mockClass)
     {
+        $propertiesCode = '';
+
+        // PHP 8.0+ : Generate promoted properties from constructor
+        if (method_exists(\ReflectionParameter::class, 'isPromoted')) {
+            $propertiesCode .= $this->generatePromotedProperties($class);
+        }
+
+        // PHP 8.4+ : Generate property hooks
+        if (method_exists(\ReflectionProperty::class, 'getHooks')) {
+            $propertiesCode .= $this->generatePropertiesWithHooks($class);
+        }
+
+        // PHP 8.4+ : Generate asymmetric visibility
+        if (method_exists(\ReflectionProperty::class, 'isPublicSet')) {
+            $propertiesCode .= $this->generatePropertiesWithAsymmetricVisibility($class);
+        }
+
+        // PHP 8.2+ : Check if class is readonly
+        $classModifiers = 'final ';
+        try {
+            if (method_exists($class, 'isReadOnly') && $class->isReadOnly()) {
+                $classModifiers .= 'readonly ';
+            }
+        } catch (\Throwable $e) {
+            // Ignore errors when checking for readonly class (e.g., mocked ReflectionClass)
+        }
```

Comment:
Mettre un try/catch ici pourrait masquer une éventuelle erreur. Il faut plutôt conditionner le test à une version la PHP, en mettant `/** @php >= 8.2 */` sur la méthode de test.

```suggestion
        if (version_compare(PHP_VERSION, '8.2.0', '>=') && $class->isReadOnly()) {
            $classModifiers .= 'readonly ';
        }
```